src/staticanalysis/bot/static_analysis_bot/workflows/local.py

```diff
@@ -64,16 +67,42 @@ def clone(self):
                                     purge=True,
                                     sharebase=settings.repo_shared_dir,
                                     branch=b'central')
-
         cmd.insert(0, hglib.HGPATH)
-        proc = hglib.util.popen(cmd)
+
+        # Start process
+        start = time.time()
+        proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+
+        # Set process stdout as non blocking
+        fcntl.fcntl(
+            proc.stdout.fileno(),
+            fcntl.F_SETFL,
+            fcntl.fcntl(proc.stdout.fileno(), fcntl.F_GETFL) | os.O_NONBLOCK,
+        )
+
+        while proc.poll() is None:
+
+            # Read and display every lines
+            out = proc.stdout.read(512)
+            if out is not None:
+                text = filter(None, out.decode('utf-8').split('\n'))
+                for line in text:
+                    logger.info('clone: {}'.format(line))
+
+            time.sleep(1)
+
         out, err = proc.communicate()
```

Comment:
This will now be half useless (err going into out)

Reply:
Yes but the call to communicate is needed to cleanly terminate the Python popen process